### ISSUE TITLE
How to update the WSL2 kernel on current Windows 10

### ISSUE THREAD

**Issue by angelog0** [microsoft/WSL#5650]
Currently,

```
$ uname -a
Linux MyPC 4.19.104-microsoft-standard #1 SMP Wed Feb 19 06:37:35 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
```

I am on current Windows 10 19041.388 and I would to know how WSL2 kernel updates. I see that `Settings Apps` has `WSL Update` package...

Will it occur with Windows Update, and I have only to wait? or is there another way?

Thanks

**Comment by koraylinux** [comment]
Before installing WSL 2, you must enable the "Virtual Machine Platform" optional feature.

Open PowerShell as Administrator and run:

```
dism.exe /online /enable-feature /featurename:VirtualMachinePlatform /all /norestart
```

Open PowerShell as Administrator and run this command to set WSL 2 as the default version when installing a new Linux distribution:

```
wsl --set-default-version 2
```

You might see this message after running that command: 

`WSL 2 requires an update to its kernel component. For information please visit https://aka.ms/wsl2kernel`. 

Please follow the link (https://aka.ms/wsl2kernel) and install the MSI from that page on our documentation to install a Linux kernel on your machine for WSL 2 to use. Once you have the kernel installed, please run the command again and it should complete successfully without showing the message.

**Comment by angelog0** [comment]
@koraylinux , I have not asked how to install WSL2. I **already** have WSL2 installed and done all steps you suggest! There are posts saying there is a new version of the kernel for WSL2, so I simply asked how to upgrade it in current version of Windows 10, if one can... _Current_ (see above) not *insider* or other development version of Windows.

**Comment by koraylinux** [comment]
Could you share the link where you seen the discussion about new kernel?

**Comment by angelog0** [comment]
[Here](https://github.com/microsoft/WSL/issues/5642), for example.

**Comment by AleCx4** [comment]
I am wondering the same thing. I have WSL2 and have made everything already to force my Ubuntu install to work with WSL2, but upon running the command: `uname -r` I get greeted with: 4.19.104-microsoft-standard, for many reasons I need 4.19.121, mainly because we can now use CUDA on WSL2 and settup a proper M.L environment on windows. Downloading the .msi installer from the official Microsoft page just updates it to version 4.19.104.

Guess we are going to have to wait for the actual update to appear on the system for itself?

**Comment by therealkenc** [comment]
> Current (see above) not insider or other development version of Windows.
> https://github.com/microsoft/WSL/issues/5642, for example.

That issue didn't state build number, but did state right at the top "Windows Insider Program with Dev Channel".

On 2004 aka 19041 the kernel is installed via https://aka.ms/wsl2kernel. Whether that msi package gets updated to a new kernel absent some (hard to image) critical bug in the kernel seems unlikely. [But I wouldn't want to jinx that with a "never".]

That CUDA is not available in Win10 2004 goes without saying.

**Comment by korypostma** [comment]
As of 2020-10-31 the msi installs 4.19.128-microsoft-standard

**Comment by PavelSosin-320** [comment]
I always tried on 2 machines - the same result: WSL2 Kernel update does nothing! According to WSL documentation page, it should be already 5.4 but it uname shows 4.9. I can't accomplish some installation procedure like runc, because it explicitly requires 5.2 at least. Everything related to running OCI containers is messed - it requires cgroups v2 and checks kernel version. Is it only uname bug or normal WSL2 Linux Kernel is only a promise?

**Comment by matheussouza9** [comment]
According to [Windows Command Line Blog](https://devblogs.microsoft.com/commandline/wsl2-will-be-generally-available-in-windows-10-version-2004/#automatic-install-and-updates), WSL2 kernel updates will be available through **Windows Update**.

I got an update, a month ago, to the [**4.19.128**](https://docs.microsoft.com/en-us/windows/wsl/kernel-release-notes#5451-microsoft-standard) version, but there is a new stable kernel release, [**5.4.51**](https://docs.microsoft.com/en-us/windows/wsl/kernel-release-notes#5451-microsoft-standard). Unfortunately, this new version isn't available yet through Windows Update (I don't know why).

![Windows Update History Screenshot](https://user-images.githubusercontent.com/6036320/100148350-6842de80-2e7b-11eb-8613-d6d9c376fd43.png)

**Comment by Biswa96** [comment]
Link to download WSL2 Linux kernel updates https://www.catalog.update.microsoft.com/Search.aspx?q=wsl

**Comment by PavelSosin-320** [comment]
The Microsoft catalog update contains 2 cab files for different architectures but  wsl --update command ignores them and reports that the 4.19 is the recent version and nothing to update. Looks like new WSL Kernels are distributed via a different channel. I tried to install the new Kernel using cab but it installs nothing. From the wsl's point of view, the cab is empty. msi installer doesn't work too - the old Kernel version is not replaced and Kernel.rollback file is not created.

**Comment by DoctorWho8** [comment]
Actually yes it does. run wsl --shutdown from an administrator upscaled window. In the location where you stored the extracted MSI kit, run that. And then restart WSL from that same window by entering wsl at the prompt and enter into its prompt the uname -a function. Restart docker if you run it as well.

**Comment by PavelSosin-320** [comment]
@DoctorWho8 Unfortunately
![wslupdate54](https://user-images.githubusercontent.com/54495912/100425398-ccda8500-3097-11eb-8402-8a56b9e5a9ac.png)
, it does nothing

**Comment by PavelSosin-320** [comment]
Will this issue be solved or the WSL stopped to work forever? I already get message 'WSL requires the Kernel update from Docker Desktop installer. I don't want toinstall 3th party installer on my machine - it looks like fishing attempt. Promised wsl --update can't find a new kernel.  
I have very fresh insider build 20231. So, I expect that somebody in MS has spent time to create the deliverable Kernel and test its installation using wsl --update..
Meanwhile I don't see that somebody is assigned.

**Comment by cpoptic** [comment]
@PavelSosin-320  +1   I too am encountering this issue.

**Comment by cpoptic** [comment]
After trying to install Windows Subsystem for Linux update using the  `wsl_update_x64.msi`, it fails to run with the error `the older version of windows subsystem for linux update cannot be removed please contact your technical support group`.

I have absolutely no idea how to fix this.  

Why can the older version not be removed???

These are the four distributions I have installed:

C:\Users\username>wsl -l -v
  NAME                   STATE           VERSION
* Ubuntu-20.04           Running         2
  Ubuntu                 Stopped         2
  docker-desktop-data    Running         2
  docker-desktop         Running         2

**Comment by PavelSosin-320** [comment]
I see something: the Kernel file is always opened by Wmwp process ( Hyper-V worker process) even after wsl --shutdown. It is a hard-to-kill process.  wsl --shutdown must stop Hyper-V worker by my understanding. Otherwise, neither update procedure will replace the Kernel file.
wsl --upgrade must ensure that Kernel file is unlocked by Wmwp before it does upgrade. But it never happens - all my distro continue to run after wsl --update attempt.
Microsoft update catalog shows 2 WSL Kernel updates, i.e. zipped cab files but how it is related to wsl --update? Can I say which update I want to install?

**Comment by angelog0** [comment]
On a tablet with win10 Home I had to revert from wsl2 to wsl1 because I had the wake up issue. Should I remove this wsl update `.msi` package?

**Comment by DoctorWho8** [comment]
Well perhaps the fact that this OS isn't part of the Insider collection might be part of the difference. Anyway I was successful following the steps I outlined above.

**Comment by CaseyLabs** [comment]
For users coming here from a Google search for WSL2 Kernel Upgrade:

- Download the latest kernel from here: https://www.catalog.update.microsoft.com/Search.aspx?q=wsl

- Open and extract the update to your desktop

- Load a command prompt with elevated privileges (Start --> cmd --> Right-click --> Run as Administrator)

- Run these commands in the command prompt:
```
cd C:\Users\your_username\Desktop\

wsl --shutdown

wsl_update_x64.msi

wsl

uname -r
```

This should update the kernel and show you the latest running version.

**Comment by cpoptic** [comment]
@CaseyLabs  I've tried this many times but it's blocked at:

> The file wsl_update_x64.msi is not a valid installation package for the product Windows Subsystem for Linux Update.  Try to find the installation package 'wsl_update_x64.msi' in a folder from which you can install Windows Subsystem for Linux Update.

Regardless of where I browse in my File Explorer and point to the downloaded `wsl_update_x64.msi` installer package, it continues to give this same response.  Blocked.

**Comment by DoctorWho8** [comment]
Right click on it and bring up the properties dialog box. Check to see if it shows that the blocked checkbox is checked. Uncheck it and close the dialog box. Then try again.

**Comment by fox4u** [comment]
Turn "Receive updates for other Microsoft products when you update Windows" on in Advance options of Windows update, the WSL2 kernel will get updated automatically.

**Comment by angelog0** [comment]
> Turn "Receive updates for other Microsoft products when you update Windows" on in Advance options of Windows update, the WSL2 kernel will get updated automatically.

This seems to fix the issue. After enabling that, it updated the kernel which now is 5.4.72!

**Comment by Biswa96** [comment]
In [Microsoft Catalog][1], the update is now 5.4.91.

[1]: https://www.catalog.update.microsoft.com/Search.aspx?q=wsl

**Comment by fenris** [comment]
any update on this ?

**Comment by TBBle** [comment]
The latest WSL 2 kernel for 20H2 19042.804 is 5.4.72.

Even if you install 5.4.91 from the Microsoft Catalog by hand, Windows Update (with "Receive updates for other Microsoft products when you update Windows" enabled) will attempt to downgrade you back to 5.4.72, which fails unless you uninstall the 5.4.91 kernel from Add/Remove Programs first. Apart from that, the hand-upgraded kernel seemed to work, but I didn't really put it through its paces.

So it seems to be working-as-intended.

@fenris What did you have in mind as an update?

I expect that Windows Insiders get Linux 5.4.91 installed by WU, but don't know for sure. The Microsoft Catalog web UI doesn't tell you which Windows builds will get which updates, so I'm not sure if there's a way to confirm this, and I don't have an Insider build running.

**Comment by ev-dev** [comment]
@TBBle I was also having this issue until I did:

1. Check installed version via `wsl --status` (mine showed 5.4.72)
2. Uncheck "Receive updates for other Microsoft products when you update Windows"
2. `wsl --shutdown`
3. Run wsl_update_x64.msi as Administrator (either via right-click in File Explorer or opening via elevated Command Prompt)
4. Open Control Panel -> Programs and Features
5. Check version info for "Windows Subsystem for Linux Update"
6. If desired update version is shown, right click "Windows Subsystem for Linux Update" and click "Repair"
7. Open new terminal/cmd prompt and again checked `wsl --status` which now shows 5.4.91 (validated in distro via `uname -r`)

Good luck!

**Comment by ev-dev** [comment]
I want to try more recent kernal versions like the 5.5.3 branch from https://github.com/microsoft/WSL2-Linux-Kernel/tree/v5.5.3 which seems to be latest release build. No prebuilt msi installer is available for these but maybe source build & kernal patch is feasible? Anyone try this?

**Comment by netcore-jroger** [comment]
> For users coming here from a Google search for WSL2 Kernel Upgrade:
> 
> * Download the latest kernel from here: https://www.catalog.update.microsoft.com/Search.aspx?q=wsl
> * Open and extract the update to your desktop
> * Load a command prompt with elevated privileges (Start --> cmd --> Right-click --> Run as Administrator)
> * Run these commands in the command prompt:
> 
> ```
> cd C:\Users\your_username\Desktop\
> 
> wsl --shutdown
> 
> wsl_update_x64.msi
> 
> wsl
> 
> uname -r
> ```
> 
> This should update the kernel and show you the latest running version.

@ev-dev

**Comment by ev-dev** [comment]
Yes @netcore-jroger this what I followed as well, but it did not work for me. 

The only piece I added was the manual "Repair" step in control panel programs manager. That triggered a blank system progress bar to show and I believe some kind of install fix which only after let me validate the latest version was active (perhaps a registry tweak or something else that wasn't being set right during install? apologies, not very savy about windows internals)

Regardless, the manual "Repair" of the install is certainly something else to try if you're having problems getting a newer kernel to install & persist

**Comment by netcore-jroger** [comment]
> Yes @netcore-jroger this what I followed as well, but it did not work for me. 
> 
> The only piece I added was the manual "Repair" step in control panel programs manager. That triggered a blank system progress bar to show and I believe some kind of install fix which only after let me validate the latest version was active (perhaps a registry tweak or something else that wasn't being set right during install? apologies, not very savy about windows internals)
> 
> Regardless, the manual "Repair" of the install is certainly something else to try if you're having problems getting a newer kernel to install & persist

I used that method.  Solved my problem.  May be the system version is different.

**Comment by Folcon** [comment]
Is access to the windows insider program required to get any updates? It seems like the latest version is quite old at this point and I can't find a viable update path...

**Comment by NoSubstitute** [comment]
There is no msi available there for kernels, only cab files.

> For users coming here from a Google search for WSL2 Kernel Upgrade:
> 
> * Download the latest kernel from here: https://www.catalog.update.microsoft.com/Search.aspx?q=wsl
> * Open and extract the update to your desktop
> * Load a command prompt with elevated privileges (Start --> cmd --> Right-click --> Run as Administrator)
> * Run these commands in the command prompt:
> 
> ```
> cd C:\Users\your_username\Desktop\
> 
> wsl --shutdown
> 
> wsl_update_x64.msi
> 
> wsl
> 
> uname -r
> ```
> 
> This should update the kernel and show you the latest running version.

**Comment by NoSubstitute** [comment]
lol, sorry. Found the msi file inside the cab.

**Comment by NoSubstitute** [comment]
> How can I stop this stupid Windows Update from installing a lower version of the kernel?

You can use the _Show or Hide Updates_ troubleshooter.
It has been removed from the official windows support pages, but is still available over on tenforums here.
https://www.tenforums.com/tutorials/8280-hide-show-windows-updates-windows-10-a.html

**Comment by jtmoon79** [comment]
**tl;dr** the kernel-in-use appears affected by the user running the WSL instance. In my csae, only the Administrator user "sees" the updated kernel.

I update a Windows 10 Pro host to [_Windows Subsystem for Linux Update - 5.10.43_](https://www.catalog.update.microsoft.com/Search.aspx?q=wsl).

1. I downloaded the `wsl_update_x64_d563b3571b8cdddd4b587266a499f98607b27ced.cab` and extracted `wsl_update_x64.msi`.
2. In a Powershell console, run as Administrator, I ran
   ```
   PS> wsl --shutdown
   PS> wsl_update_x64.msi
   PS> wsl -- uname -a
   ```
   The GUI install appeared to succeed.
   The last `uname` command printed the updated kernel.

3. As Administrator, I ran `wsl --shutdown`.

4. In a Powershell console, run as my non-Administrator user and not "main user" (not the user that installed WSL under elevated privileges), I repeated the prior commands.
**The `uname` command, run as a non-privileged uesr, displayed the old kernel version.**

**Comment by Calusin** [comment]
I had the same issue and I solved it by downloading the latest kernel from this site ♥

> Link to download WSL2 Linux kernel updates https://www.catalog.update.microsoft.com/Search.aspx?q=wsl

**Comment by xDFCx** [comment]
Hi @jtmoon79, looks like I have the same thing here

> wsl --status
...
Kernel version: 5.10.60.1
> wsl -- uname -r
4.4.0-19041-Microsoft

Have you figured out anything on this issue? Is it possible to update kernet for my user?

**Comment by git-adamk** [comment]
I had similar issue. In my case it was caused because I had WSL in version 1.
Open cmd as admin and type:
wsl --list --verbose
  NAME            STATE           VERSION
* Ubuntu-18.04    Running         2

if version is 1 change it to 2 by typing:
wsl --set-version <distribution name> 2

After that wsl --status in cmd and uname -r in WSL shows the same kernel version.

**Comment by Abe-Telo** [comment]
now why is this not working?
![image](https://user-images.githubusercontent.com/29134216/156618542-790f9c81-658c-49b2-958b-632628baf976.png)

**Comment by morgan-greywolf** [comment]
On Thu, 2022-03-03 at 09:26 -0800, thething wrote:
> now why is this not working?
> 
Exactly what the error message states: the sources for kernel stated
can't be found.  You may have to point the build toward the sources if
they can't be found automatically, either through ./configure options
or through flags passed directly to make.

**Comment by stloc** [comment]
on ubuntu upgrade 

> " Newer kernel available                                                                                        
>                                                                                                                
>    The currently running kernel version is 5.10.102.1-microsoft-standard-WSL2 which is not the expected kernel  
>    version 5.15.0-40-lowlatency.                                                                                 
>                                                                                                                 
>    Restarting the system to load the new kernel will not be handled automatically, so you should consider       
>    rebooting."

![image](https://user-images.githubusercontent.com/4832070/178011077-1f9b0c2b-f2a2-49dc-af7c-dbdfa5536ff8.png)

how upgrade kernel ?

**Comment by TBBle** [comment]
@stloc Depending on what version of Windows you're on, `wsl --update` should work to upgrade the kernel if a new kernel is available. The kernel inside Ubuntu (i.e. the one that notice is talking about) is not used in WSL2.

**Comment by NaveedAhmadHematmal** [comment]
> Link to download WSL2 Linux kernel updates https://www.catalog.update.microsoft.com/Search.aspx?q=wsl

This link fixed my problem. 👍

**Comment by rsokolewicz** [comment]
Similar problem here. According to wsl I have the latest kernel installed, according to the linux distro I don't:

```
> wsl --status
Default Distribution: Ubuntu-20.04
Default Version: 2

Windows Subsystem for Linux was last updated on 09/10/2022
WSL automatic updates are on.

Kernel version: 5.10.102.1
```

vs

```
> wsl -- uname -r
4.4.0-19041-Microsoft
```

I've updated wsl from 1 to 2, tried reinstalling the kernel update, running the repair tool. All doesn't solve the problem. Does anyone have other ideas?

**Comment by CaseyLabs** [comment]
@rsokolewicz - it sounds like your Ubuntu installation is still running under WSL1. 

Even though you've set the WSL Default Version to 2, that setting doesn't get applied to machines that were originally created under WSL1.

You can upgrade your Ubuntu distribution to WSL2 by running:

```
wsl --set-version Ubuntu-22.04 2
```

See here for more details:
https://superuser.com/questions/1725627/which-linux-kernel-do-i-have-in-wsl
https://superuser.com/questions/1725627/which-linux-kernel-do-i-have-in-wsl

**Comment by rsokolewicz** [comment]
hi @CaseyLabs, that worked for me. thanks :)

**Comment by bluebrown** [comment]
This page doesn't show the 5.15 kernel. https://www.catalog.update.microsoft.com/Search.aspx?q=wsl. If I try `wsl --update`, I don't get it either.

```
$ wsl --status
Default Version: 2

Windows Subsystem for Linux was last updated on 04-Nov-22
The Windows Subsystem for Linux kernel can be manually updated with 'wsl --update', but automatic updates cannot occur due to your system settings.
To receive automatic kernel updates, please enable the Windows Update setting: 'Receive updates for other Microsoft products when you update Windows'.
For more information please visit https://aka.ms/wsl2kernel.

Kernel version: 5.10.102.1
```


```
$ wsl --update
Checking for updates...
The Windows Subsystem for Linux kernel can be manually updated with 'wsl --update', but automatic updates cannot occur due to your system settings.
To receive automatic kernel updates, please enable the Windows Update setting: 'Receive updates for other Microsoft products when you update Windows'.
For more information please visit https://aka.ms/wsl2kernel.
```


I cannot enable automatic update since its managed by my org. So I want to install 5.15 kernel manually. How to do that?

**Comment by Biswa96** [comment]
Updated Linux kernel is provided with WSL package https://apps.microsoft.com/store/detail/windows-subsystem-for-linux/9P9TQF7MRM4R?hl=en-us&gl=us. Can you install that?

**Comment by bluebrown** [comment]
@Biswa96 , is there a install without the store? I cant install stuff from the store for some reason.

**Comment by Biswa96** [comment]
> is there a install without the store? 

Sure. Just download the msixbundle file from here https://github.com/microsoft/WSL/releases and install it with `Add-AppxPackage` command in Powershell. Remember that it requires certain versions of Windows 10 or 11 OSes which can be learned from [here](https://devblogs.microsoft.com/commandline/the-windows-subsystem-for-linux-in-the-microsoft-store-is-now-generally-available-on-windows-10-and-11/).

**Comment by bluebrown** [comment]
@Biswa96, OK, thanks. Unfortunately, this didn't change the kernel verion it seems. I have removed the old kernel update previously. Then from admin pwsh

```
PS C:\Users\00300201\Downloads>  Add-AppxPackage C:\Users\00300201\Downloads\Microsoft.WSL_1.0.3.0_x64_ARM64.msixbundle
```

The output here is in german but It said there is no kernel, and then I did wsl update

```
 wsl --status
Standardversion: 2

Der Windows-Subsystem für Linux-Kernel kann mit „wsl --update“ manuell aktualisiert werden. Aufgrund Ihrer Systemeinstellungen können jedoch keine automatischen Updates durchgeführt werden.
Um automatischen Kernel-Updates zu erhalten, aktivieren Sie die Windows Update-Einstellung: „Empfangen von Updates für andere Microsoft-Produkte, wenn Windows aktualisiert wird“.
Weitere Informationen finden Sie unter https://aka.ms/wsl2kernel.

Die WSL 2-Kerneldatei wird nicht gefunden. Wenn Sie den Kernel aktualisieren oder wiederherstellen möchten, führen Sie bitte das "wsl.exe--Update" aus.
PS C:\Users\00300201\Downloads> wsl --update
Es wird nach Updates gesucht...
Updates werden heruntergeladen...
Updates werden installiert...
Diese Änderung wird beim nächsten vollständigen Neustart von WSL wirksam. Führen Sie „wsl--shutdown“ aus, um einen Neustart zu erzwingen.
Kernel-Version: 5.10.102.1
```

And the gernel is still `5.10.102.1`

```
 wsl --status
Default Version: 2

Windows Subsystem for Linux was last updated on 23-Dec-22
The Windows Subsystem for Linux kernel can be manually updated with 'wsl --update', but automatic updates cannot occur due to your system settings.
To receive automatic kernel updates, please enable the Windows Update setting: 'Receive updates for other Microsoft products when you update Windows'.
For more information please visit https://aka.ms/wsl2kernel.

Kernel version: 5.10.102.1
```

My windows version should be ok for this

```
Edition	Windows 10 Enterprise
Version	22H2
Installed on	‎21-‎Jun-‎21
OS build	19045.2251
Experience	Windows Feature Experience Pack 120.2212.4180.0
```

**Comment by Biswa96** [comment]
Do you have any `kernel` option in `%UserProfile%\.wslconfig` file ?

**Comment by bluebrown** [comment]
@Biswa96 , no. I dont have a .wlsconfig at all in my profile folder. I also checked the admin users profile folder.

**Comment by bluebrown** [comment]
It's also suspicious that there is no new options for the --install command. As far as I know there should be a `--pre-release` flag and also one thats called `--web-install` or something to install from github.

My options still show only this:

```
 --install [Options]
        Install additional Windows Subsystem for Linux distributions.
        For a list of valid distributions, use 'wsl --list --online'.

        Options:
            --distribution, -d [Argument]
                Downloads and installs a distribution by name.

                Arguments:
                    A valid distribution name (not case sensitive).

                Examples:
                    wsl --install -d Ubuntu
                    wsl --install --distribution Debian
```

**Comment by degarb** [comment]
wsl --set-default-version 2 fails,
and,
updating msi just hangs.     ( I have done everything on, https://dockerwebdev.com/tutorials/install-docker/)  (One reason I use windows is because instructions for anything linux never work, unlike windows.  I have been an on off linux user since 1999.)

**Comment by degarb** [comment]
Also, repos never work for me.  My MS store, works about 10 percent of the time.  So, I can't install ubuntu from there either.

**Comment by codingexcuse** [comment]
wsl --update just run this it worked for me

**Comment by Li-00** [comment]
My environment is win11 i5 64-bit
Use the administrator to open powerShell
  first ,wsl --udate,
 seconed check version   wsl --version    core > 5 
![image](https://github.com/microsoft/WSL/assets/71139000/0cf4bcf7-75d6-40cd-a56e-07fef0a86a60)

**Comment by TBBle** [comment]
The latest 5.15 series kernel release is 5.15.90.1. There's [a 6.1 series release](https://github.com/microsoft/WSL2-Linux-Kernel/releases) but I'm not sure why it's not yet being distributed (might still be WIP?).

If you're stuck on 5.10.102.2 or older, that's the latest release [from the Microsoft Update Catalog](https://www.catalog.update.microsoft.com/Search.aspx?q=Windows%20Subsystem%20for%20Linux), I suspect in this case one needs to switch to the Microsoft Store version of WSL to get newer kernels.

Looking at https://github.com/microsoft/WSL/releases, the 1.3 series of the Windows Store version of WSL includes Linux 5.15.90.2-3, so I'm assuming the 6.1 series is not ready yet. (That version doesn't appear on the WSL2-Linux-Kernel repo AFAICT)

![image](https://github.com/microsoft/WSL/assets/138397/9eed66d8-84ef-4dfe-8e94-e05aa847c63e)

Anyway, since this ticket predates the Microsoft Store move, I'd suggest anyone having problems with updates after 5.10.102.2 open a new ticket, with full details, _particularly_ `wsl --version` output, as this ticket is closed and probably not being looked at by many people.

**Comment by TaduJR** [comment]
The Microsoft Community wrote an article for upgrading to Kernel V6. Check the link below.
https://learn.microsoft.com/en-us/community/content/wsl-user-msft-kernel-v6
